cumulusci/utils.py

```diff
@@ -181,18 +181,51 @@ def zip_subfolder(zip_src, path):
     return zip_dest
 
 
-def zip_inject_namespace(
-    zip_src,
+def process_text_in_directory(path, process_file):
+    for path, dirs, files in os.walk(path):
+        for orig_name in files:
+            orig_path = os.path.join(path, orig_name)
+            try:
+                with open(orig_path, "r", encoding="utf-8") as f:
+                    orig_content = f.read()
+            except UnicodeDecodeError:
+                # Probably a binary file; skip it
+                continue
+            new_name, new_content = process_file(orig_name, orig_content)
+            new_path = os.path.join(path, new_name)
+            if new_name != orig_name:
+                os.rename(orig_path, new_path)
+            with open(new_path, "w", encoding="utf-8") as f:
+                f.write(new_content)
+
+
+def process_text_in_zipfile(zf, process_file):
+    new_zf = zipfile.ZipFile(io.BytesIO(), "w", zipfile.ZIP_DEFLATED)
+    for name in zf.namelist():
+        content = zf.read(name)
+        try:
+            content = content.decode("utf-8")
+        except UnicodeDecodeError:
+            # Probably a binary file; don't change it
+            pass
+        else:
+            name, content = process_file(name, content)
+        new_zf.writestr(name, content)
+    return new_zf
+
+
+def inject_namespace(
+    name,
+    content,
     namespace=None,
     managed=None,
     filename_token=None,
     namespace_token=None,
     namespaced_org=None,
     logger=None,
 ):
-    """ Replaces %%%NAMESPACE%%% for all files and ___NAMESPACE___ in all
-        filenames in the zip with the either '' if no namespace is provided
-        or 'namespace__' if provided.
+    """ Replaces %%%NAMESPACE%%% in file content and ___NAMESPACE___ in filename
```

Comment:
```suggestion
    """ Replaces %%%NAMESPACE%%% in file content and ___NAMESPACE___ in name
```
Did you mean 'name' instead of 'filename'? It's not apparent to me what 'filename' refers to here.

Reply:
The `name` arg contains the filename that `content` came from.